src/client.js

```diff
@@ -1654,16 +1654,41 @@ class Client {
     async getSchema(schemaId, representation, internal) {
         var entity = await this._entityCache.get("xtk:schema", schemaId);
         if (!entity) {
-          entity = await this.getEntityIfMoreRecent("xtk:schema", schemaId, "xml", internal);
-          if (entity) {
-            const impls = DomUtil.getAttributeAsString(entity, "implements");
-            if (impls === "xtk:persist" && schemaId !== "xtk:session" && schemaId !== "xtk:persist") {
-                // Ensure xtk:persist is present by loading the xtk:session schema
-                await this.getSchema("xtk:session", "xml", true);
+              // special case of temp schema for nms:group
+              if (schemaId.startsWith("temp:group:")) {
+                  var parts = schemaId.split(":");
+                  var queryDef = {
+                    "schema": "nms:group",
+                    "operation": "get",
+                    "select": {
+                      "node": [
+                        { "expr": "@id" },
+                        { "expr": "extension" }
+                      ]
+                    },
+                    "where": {
+                      "condition": [
+                        { "expr": "@id=" + parts[2] }
```

Comment:
This is a potential xtk injection my friend. You may want to check that parts[2] is a valid number. You can use XtkCaster.asLong for example. Even if the injection is theoretical, it's important to ALWAYS escape or secure the input parameters of public functions